src/backend/joanie/core/serializers.py

```diff
@@ -77,14 +77,15 @@ def get_is_graded(self, target_course):
         product = self.context.get("product", None)
         return target_course.product_relations.get(product=product).is_graded
 
-    @staticmethod
-    def get_course_runs(target_course):
+    def get_course_runs(self, target_course):
         """
         Return related course runs ordered by start date asc
         """
-        return CourseRunSerializer(
-            target_course.course_runs.order_by("start"), many=True
-        ).data
+        product = self.context.get("product", None)
+        course_runs = product.target_course_runs.filter(course=target_course).order_by(
```

Comment:
this will fail if product is None. Add a test?

Reply:
Oh good catch, there is also this case to get `is_graded` and `position` attributes